### ISSUE TITLE
docker: "network disconnect" requires 2 arguments.

### ISSUE THREAD

**Issue by derwaldgeist** [zodern/meteor-up#704]
Since today, I am getting this error message when mup is trying to perform the "Start Meteor" task:

```
x Start Meteor: FAILED
	
	-----------------------------------STDERR-----------------------------------
	-letsencrypt
	Error: failed to remove containers: [guzz-nginx-letsencrypt]
	docker: "network disconnect" requires 2 arguments.
	See 'docker network disconnect --help'.
	
	Usage:	docker network disconnect [OPTIONS] NETWORK CONTAINER
	
	Disconnects container from a network
	Error response from daemon: no such id: guzz-nginx-proxy
	Error: failed to remove containers: [guzz-nginx-proxy]
	docker: "network disconnect" requires 2 arguments.
	See 'docker network disconnect --help'.
	
	Usage:	docker network disconnect [OPTIONS] NETWORK CONTAINER
	
	Disconnects container from a network
	Finished removing docker containers
	--2017-08-23 12:43:41--  https://raw.githubusercontent.com/jwilder/nginx-proxy/master/nginx.tmpl
	Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.60.133
	Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.60.133|:443... connected.
	HTTP request sent, awaiting response... 503 first byte timeout
	2017-08-23 12:44:09 ERROR 503: first byte timeout.
	
	-----------------------------------STDOUT-----------------------------------
	guzz
	base: Pulling from abernix/meteord
	Digest: sha256:17d679c104d54c92e10a6703dc9f0951747f9041adc3eab25989a1fb576a420c
	Status: Image is up to date for abernix/meteord:base
	Pulled abernix/meteord:base
	2198f594acd69d2561de7236d2dd223c37bdbf632505321d98a86bd3b162ec30
	Ran abernix/meteord:base
	Running autogenerate
	----------------------------------------------------------------------------
```

**Comment by zodern** [comment]
https://status.github.com/  raw.githubusercontent.com isn't working right now.

I might modify it to continue if the file is already downloaded to prevent this from happening in the future.

**Comment by derwaldgeist** [comment]
OK, thanks for the quick response!

Just FYI: I tried again using the most recent mup version 1.3.0. It ran through with `Verifying Deployment: SUCCESS`, but the app wasn't started dispite of this. When I tried to show the logs via `mup logs`, I got this error message in return:

```
Error response from daemon: can not get logs from container which is dead or marked for removal
```

Wondering why mup deploy returned with `Verifying Deployment: SUCCESS` despite of this error?

And is there a way to get rid of this "dead or marked for removal" problem?

**Comment by zodern** [comment]
`Verifying Deployment` currently doesn't work when using Let's Encrypt or the reverse proxy.

What does `mup docker ps -a` show?

**Comment by derwaldgeist** [comment]
Seems as if my app's container is dead, whatever that means:
```
CONTAINER ID        IMAGE                                           COMMAND                  CREATED              STATUS              PORTS                                      NAMES
128cebc490db        jrcs/letsencrypt-nginx-proxy-companion:latest   "/bin/bash /app/en..."   45 seconds ago       Up 44 seconds                                                  guzz-nginx-letsencrypt
1996ba056c75        jwilder/nginx-proxy:latest                      "/app/docker-entry..."   About a minute ago   Up About a minute   0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   guzz-nginx-proxy
2198f594acd6        66e350aaea2e                                    "/bin/sh -c 'bash ..."   About an hour ago    Dead                                                           guzz
3c37c89bd61c        mongo                                           "/entrypoint.sh mo..."   7 months ago         Up 38 minutes       127.0.0.1:27017->27017/tcp                 mongodb
```

**Comment by derwaldgeist** [comment]
Is there a way to reset everything? I tried `mup setup` multiple times, but this didn't fix it. Also tried `mup stop` followed by `mup setup` and `mup deploy`, but the container remains in its dead state. Would be great if there was a kind of `mup uninstall` that cleans up everything.

**Comment by zodern** [comment]
You could reinstall/recreate the server and run `mup setup` and `mup deploy`.
You could also look at https://stackoverflow.com/questions/30794108/docker-cannot-remove-dead-container to try to remove the container.

**Comment by derwaldgeist** [comment]
Thanks. Already found a similar thread and tried to remove the image using `docker rm -f`. But this didn't work. I got this error message:

```
Error response from daemon: driver "aufs" failed to remove root filesystem for 2198f594acd69d2561de7236d2dd223c37bdbf632505321d98a86bd3b162ec30: error preparing atomic delete of aufs mountpoint for id: 2198f594acd69d2561de7236d2dd223c37bdbf632505321d98a86bd3b162ec30: rename /var/lib/docker/aufs/mnt/2198f594acd69d2561de7236d2dd223c37bdbf632505321d98a86bd3b162ec30 /var/lib/docker/aufs/mnt/2198f594acd69d2561de7236d2dd223c37bdbf632505321d98a86bd3b162ec30-removing: device or resource busy
```
Maybe it's related to the fact that my server is using 96% of its free disk space? Will try the file system based `rm -r` now. Is there a way to stop docker completely before doing such stuff?

**Comment by zodern** [comment]
You can stop docker with `sudo service docker stop`.

**Comment by derwaldgeist** [comment]
Thanks. Cleaned up all images but the MongoDB instance now. Let's see what happens on deploy.

**Comment by derwaldgeist** [comment]
Ok, the server is up and running again. Still wondering what caused the initial error message and broke the container, though. Shall I keep this open?

**Comment by zodern** [comment]
If you run `mup start --verbose`, does it still show `docker: "network disconnect" requires 2 arguments.`?

For the error from downloading `https://raw.githubusercontent.com/jwilder/nginx-proxy/master/nginx.tmpl`, [it should be working again](https://status.github.com/messages). The reverse proxy, which will replace `meteor.ssl`, won't have the error.

**Comment by derwaldgeist** [comment]
Here's the output:
```
using image
removing last
Image mup-guzz:latest
Volume
Removing docker containers. Errors about nonexistent endpoints and containers are normal.
Error response from daemon: No such container: guzz
Error response from daemon: endpoint guzz not found
Error response from daemon: No such container: guzz-frontend
Error response from daemon: endpoint guzz-frontend not found
Error response from daemon: No such container: guzz-nginx-letsencrypt
Error response from daemon: endpoint guzz-nginx-letsencrypt not found
Error response from daemon: No such container: guzz-nginx-proxy
Error response from daemon: endpoint guzz-nginx-proxy not found
Finished removing docker containers
7575a28d32c816613704f80a1f322e7b10dd0f51bd726f2fa32e9e67f5be856b
Ran abernix/meteord:base
Running autogenerate
--2017-08-23 17:15:14--  https://raw.githubusercontent.com/jwilder/nginx-proxy/master/nginx.tmpl
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.60.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.60.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 10529 (10K) [text/plain]
Saving to: ‘/opt/guzz/config/nginx.tmpl’

     0K ..........                                            100% 23.5M=0s

2017-08-23 17:15:14 (23.5 MB/s) - ‘/opt/guzz/config/nginx.tmpl’ saved [10529/10529]

latest: Pulling from jrcs/letsencrypt-nginx-proxy-companion
Digest: sha256:de2a41c7f99c6a942769d43162dce20abcd004b624c5b86b61073a5c536b313f
Status: Image is up to date for jrcs/letsencrypt-nginx-proxy-companion:latest
latest: Pulling from jwilder/nginx-proxy
Digest: sha256:f2a37de9e65a20a1119cc7d8e98c9cf39dd4916b4a0f5f7eba9ddcb35290c83b
Status: Image is up to date for jwilder/nginx-proxy:latest
Pulled autogenerate images
da84ca6f084e39669bd57135d87f121ecb4ee86f498da4642f9cae8e7c0cfe02
Ran nginx-proxy
132bb40469a03e3756d86a9300c83a312c8089257849a06e890f52cc28572e0b
Ran jrcs/letsencrypt-nginx-proxy-companion
```

**Comment by elie222** [comment]
Have this same issue. How did you end up solving it?

Got to this:
``` Error response from daemon: driver "aufs" failed to remove root filesystem for 2198f594acd69d2561de7236d2dd223c37bdbf632505321d98a86bd3b162ec30: error preparing atomic delete of aufs mountpoint for id: ```

**Comment by derwaldgeist** [comment]
I had to manually delete the containers. But this only worked if I restarted the server after the containers were marked for deletion. Otherwise, I got the error that they couldn't be removed from disk.

The steps are like this:

* `docker ps -a` to show all containers
* `docker rm -f <container-id>` for all containers you want to delete
* `sudo reboot` to restart the machine
* `docker system prune` to cleanup the remainders of the dead containers on disk
* `mup deploy`
